devicetypes/vision-raytseng/vision-4-in-1-motion-sensor.src/vision-4-in-1-motion-sensor.groovy

```diff
@@ -0,0 +1,495 @@
+	parameter.parameter5description = ""
+	parameter.parameter5type = "number"
+	parameter.parameter5options = "1..127"
+	parameter.parameter5enumMap = []
+	parameter.parameter5default = 3
+	parameter.parameter5size = 1
+	
+	parameter.parameter6name = "MotionSensitivity"
+	parameter.parameter6title = "Motion active sensitivity"
+	parameter.parameter6description = ""
+	parameter.parameter6type = "enum"
+	parameter.parameter6options = ["Highest", "Higher", "High", "Medium", "Low", "Lower", "Lowest"]
+	parameter.parameter6enumMap = ["Highest": 1, "Higher": 2, "High": 3, "Medium": 4, "Low": 5, "Lower": 6, "Lowest": 7]
+	parameter.parameter6default = "Medium"
+	parameter.parameter6size = 1
+	
+	parameter.parameter7name = "LedDispMode"
+	parameter.parameter7title = "LED display mode"
+	parameter.parameter7description = ""
+	parameter.parameter7type = "enum"
+	parameter.parameter7options = ["LED off when Temperature report/Motion active",
+		"LED blink when Temperature report/Motion active",
+		"LED blink when Motion active/LED off when Temperature report"]
+	parameter.parameter7enumMap = ["LED off when Temperature report/Motion active": 1,
+		"LED blink when Temperature report/Motion active": 2,
+		"LED blink when Motion active/LED off when Temperature report": 3]
+	parameter.parameter7default = "LED off when Temperature report/Motion active"
+	parameter.parameter7size = 1
+	
+	parameter.parameter8name = "RetryTimes"
+	parameter.parameter8title = "Motion notification retry times"
+	parameter.parameter8description = ""
+	parameter.parameter8type = "number"
+	parameter.parameter8options = "0..10"
+	parameter.parameter8enumMap = []
+	parameter.parameter8default = 3
+	parameter.parameter8size = 1
+	
+	return parameter."parameter${num}${text}"
+}
+
+def wakeUpIntervalUpdate(interval) {
+	def value = interval?.find { it.key == "${getWakeUpInfo("name")}" }?.value
+	
+	if (value != null) {
+		if (state."${getWakeUpInfo("name")}" != value) {
+			state."${getWakeUpInfo("name")}" = value
+			state."${getWakeUpInfo("name")}Refresh" = true
+		}
+	}
+}
+
+def wakeUpIntervalCheck(interval) {
+	def value = interval?.find { it.key == "${getWakeUpInfo("name")}" }?.value
+	
+	if (value != null) {
+		if (state."${getWakeUpInfo("name")}" == value) {
+			if (state."${getWakeUpInfo("name")}Refresh" == true) {		
+				state."${getWakeUpInfo("name")}Refresh" = false
+			}
+		}
+	}
+}
+
+def getWakeUpInfo(text) {
+	def wakeUp = [:]
+	
+	wakeUp.name = "wakeUpInterval"
+	wakeUp.title = "Wake up interval [Hours]"
+	wakeUp.description = ""
+	wakeUp.type = "number"
+	wakeUp.range = "1..4660"
+	wakeUp.default = 24
+	
+	return wakeUp."${text}"
+}
+
+private getCommandClassVersions() {
+	[
+		0x5E: 2,
+		0x22: 1,
+		0x85: 2,
+		0x59: 3,
+		0x80: 1,
+		0x70: 2,
+		0x5A: 1,
+		0x7A: 5,
+		0x87: 3,
+		0x72: 2,
+		0x8E: 3,
+		0X31: 5,
+		0x71: 3,
+		0x73: 1,
+		0x98: 1,
+		0x9F: 1,
+		0x6C: 1,
+		0x55: 2,
+		0x86: 3,
+		0x84: 2 
```

Comment:
you do not need to include all of these, only the ones whose messages you are parsing/handling

Reply:
Okay, I fixed it. Now I only include these command version that I parse and handle.